addons/web/static/src/views/fields/x2many/x2many_field.js

```diff
@@ -258,9 +270,20 @@ export const x2ManyField = {
     displayName: _lt("Relational table"),
     supportedTypes: ["one2many", "many2many"],
     useSubView: true,
-    extractProps: ({ attrs }) => ({
-        addLabel: attrs["add-label"],
-    }),
+    extractProps: ({ attrs, viewMode, views, widget, options, string }) => {
+        const props = {
+            addLabel: attrs["add-label"],
+            viewMode,
+            views,
+            archInfo: viewMode && views[viewMode],
+            options,
```

Comment:
question: do we want to split those options to make direct props instead?

Reply:
["create", "createEdit", "delete", "link", "unlink", "write"];